### PR TITLE
Convert dynamic parser require to static

### PR DESCRIPTION
**Pull request by MisterMX**
### Description

This PR aims to convert the dynamic require statements in `src/parser/typeRegistry.js` to static ones. So the package can be used with Webpack which requires static imports.

This does not work out of the box because this creates a circular dependency between `src/parser/element.js` and `src/parser/xsd/xsdElement.js`. To fix this, the `typeRegistry` is removed from `Element` and instead passed as function parameter when it is needed.

This PR does not add any additional functionalilty. It just changes the way dependencies are loaded.

#### Related issues

Currently, using this `strong-soap` with Webpack does not work (see #109).

### Checklist

- [X] Make dynamic imports static
- [X] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by avindra** [comment]
@MisterMX what's the use case for webpack? Isn't it for bundling applications in the browser? Whereas `strong-soap` is intended for use in the node.js environment.

**Comment by MisterMX** [comment]
@avindra we use Webpack on our backend side to minimize the size of the Docker image. This allows us to reduce the size of the project (> 200 MB for the node_modules folder) to > 10 MB (the single bundle.js).

As a nice side effect, we also noticed a faster startup time since only one JS file needs to be parsed.

**Comment by avindra** [comment]
@MisterMX thanks for the clarification, that is news to me. I used Webpack in the beginning when it was oriented at building browser apps. Very interesting to hear about this use case.

The dynamic require statements are probably not very compatible with other things too, I think this is a good change.

File: src/parser/element.js [hunk 1]
```diff
@@ -8,7 +8,6 @@
 var g = require('../globalize');
 var assert = require('assert');
 var QName = require('./qname');
-var typeRegistry = require('./typeRegistry');
```
Comment (deepakrkris) [inline src/parser/element.js]: any reason we choose to pass type registry through wsdl.js and not load here ?

File: src/parser/element.js [hunk 2]
```diff
@@ -91,16 +90,12 @@ class Element {
       debug('Element %s is not allowed within %j', qname, this.nsName);
     }
     
-    if (ElementType) {
```
Comment (deepakrkris) [inline src/parser/element.js]: is ElementType expected to be returned always ?

File: src/parser/element.js [hunk 3]
```diff
@@ -91,16 +90,12 @@ class Element {
...
-      child = new ElementType(nsName, attrs, options);
-      child.nsURI = qname.nsURI;
-      child.targetNamespace = attrs.targetNamespace || this.getTargetNamespace();
-      debug('Element created: ', child);
-      child.parent = parent;
-      stack.push(child);
-    } else {
-      this.unexpected(nsName);
```
Comment (deepakrkris) [inline src/parser/element.js]: removing this error block seems to be unrelated to this change

File: src/parser/element.js (hunk 1 above)
Comment (MisterMX) [reply src/parser/element.js]: This does not work, because it introduces a circular dependency: `element.js` → `typeRegistry.js` → `element.js`

File: src/parser/element.js (hunk 2 above)
Comment (MisterMX) [reply src/parser/element.js]: As far as I understood the code, the function `getElementType` in `typeRegistry.js` never returns `null` or `undefined`. So there is no need to check for it.

File: src/parser/element.js (hunk 3 above)
Comment (MisterMX) [reply src/parser/element.js]: That's true...

Unfortunately, because my change reduces the test coverage very slightly, which causes Travis CI to fail.
So I was forced to remove some lines somewhere in the code to get it into positive.

If you have a better solution for this issue, feel free to suggest a change. Since I am not very deep into the code I cannot do that.

**Comment by MisterMX** [comment]
@deepakrkris I added some answers to your comments. If you have a better solution, feel free to suggest a change.

**Comment by codeitcody** [comment]
We use webpack to package our backend code for deployment on AWS Lambda (due to strict size requirements) so this PR would help our use case as well.
